File: JCleanArchitectureApp/data/src/main/java/com/woowrale/data/repository/local/LocalRepository.java
package com.woowrale.data.repository.local;

import com.woowrale.domain.model.Contact;

import java.util.List;

public class LocalRepository {
    private LocalContactSource localContactSource;

    public LocalRepository(LocalContactSource localContactSource){
        this.localContactSource = localContactSource;
    }

    public Contact findById(Integer id){
        return localContactSource.findById(id);
    }

    public List<Contact> getAll(){
        return localContactSource.getPopularContacts();
    }

    public void saveAll(List<Contact> contacts){
        localContactSource.saveContacts(contacts);
    }

    public void updateContact(Contact contact){
        localContactSource.update(contact);
    }
}

File: KCleanArchitectureApp/data/src/main/java/com/woowrale/data/repository/local/LocalRepository.kt
package com.woowrale.data.repository.local

import com.woowrale.domain.model.Contact

interface LocalContactSource {

    fun getContacts(): List<Contact>

    fun isEmpty(): Boolean

    fun save(contacts: List<Contact>)

    fun findById(id: Int): Contact

    fun update(contact: Contact)
}

class LocalRepository(private val localContactSource: LocalContactSource) {
    fun getContacts(): List<Contact> {
        return localContactSource.getContacts()
    }

    fun isEmpty(): Boolean {
        return localContactSource.isEmpty()
    }

    fun save(contacts: List<Contact>){
        localContactSource.save(contacts)
    }

    fun findById(id: Int): Contact{
        return localContactSource.findById(id)
    }

    fun update(contact: Contact){
        localContactSource.update(contact)
    }


}